mantidproject/mantid
# Add Only update vanadium checkbox to EngDiff calibration tab

**Description of work.**

Add OnlyUpdateVanadium checkbox to skip calibration if checked
    
Only_Update_vanadium checkbox is enabled if Create_New_calibration is enabled
and the last calibration was successful.
When Only_update_vanadium is checked, crop is disabled and unchecked,
Sample path is disabled, its path not validated,
and the Main action button is renamed to "Update Vanadium"


**To test:**

- run `git test-pr origin 32184`
- Open Workbench and the Engineering Diffraction interface

_Enabling_
- The Only Update Vanadium checkbox should be disabled (greyed out)
- For Vanadium Path enter: .`../cycle_19_1/ENGINX00307521.nxs` or your favourite dataset
- For Sample Path enter: `.../cycle_19_1/ENGINX00305738.nxs` or your favourite dataset
- make sure Plot Calibrated workspace is ticked
- Then click Calibrate
- Screenshot or just keep open the plots that appear
- After it runs successfully, the Only Update Vanadium checkbox should be enabled

_Ticking/Checking_
- Tick Crop Calibration
- Now tick Only Update Vanadium and notice that this has:
    - Disabled Sample Path
    - Disabled and unticked Crop Calibration
    - The main action button is renamed from "Calibrate" to "Update Vanadium"
- The enabled/disabled state of these inputs persist after toggling to Load Existing Calibration and then toggling back to Create New Calibration

_Validation and Behaviour_
- Firstly, untick OnlyUpdateVanadium, and change the Sample path to another dataset, such as: `.../cycle_12_4/ENGINX00193749.nxs`
- Now retick Only Update Vanadium and click the main action button "Update Vanadium"
- The output plots should be the same as before!!
- Now untick OnlyUpdateVanadium and click Calibrate
- The newly output plot should be different, which proves that the sample path was ignored when OnlyUpdateVanadium was ticked.

Questions for tester:
- **Should the calibration be replotted if Only_update_vanadium is ticked? (calibration should not change!)** -> I've decided Yes!
- Are there any other tests that should be added?

Fixes #31904 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
